R/sampler-random-walk.R

```diff
@@ -22,16 +22,38 @@
 ##'
 ##' The initial point selected will lie within the domain, as this is
 ##' enforced by [monty_sample].
+##' 
+##' @param rerun_every Optional integer giving the frequency at which
+##'   we should rerun the model on the current "accepted" parameters to 
+##'   obtain a new density for stochastic models.  The default for this 
+##'   (`Inf`) will never trigger a rerun, but if you set to 100, then
+##'   every 100 steps we run the model on both the proposed *and* previously
+##'   accepted parameters before doing the comparison.  This may help "unstick"
+##'   chains, at the cost of some bias in the results.
+##'
+##' @param rerun_random Logical, controlling the behaviour of
+##'   rerunning (when `rerun_every` is finite). The default value of
+##'   `FALSE` will rerun the model at fixed intervals of iterations
+##'   (given by `rerun_every`). If `TRUE`, then we stochastically
+##'   rerun each step with probability of `1 / rerun_every`. This gives
+##'   the same expected number of MCMC steps between reruns but a different
+##'   pattern.
 ##'
 ##' @return A `monty_sampler` object, which can be used with
 ##'   [monty_sample]
 ##'
 ##' @export
-monty_sampler_random_walk <- function(vcv = NULL, boundaries = "reflect") {
+monty_sampler_random_walk <- function(vcv = NULL, boundaries = "reflect",
+                                      rerun_every = Inf, rerun_random = FALSE) {
```

Comment:
Do you want to make `rerun_random = TRUE` the default? I feel that's what most people preferred in the end